mne/io/edf/edf.py

```diff
@@ -66,6 +68,21 @@ def find_edf_events(raw):
     return raw.find_edf_events()
 
 
+def _edf_events_from_annotations(raw, event_id):
+    """Modify events_from_annotaitons for EDF specifics.
+
+    Modify events_from_annotaitons so that events[:,1] corresponds to
+    the duration of the events instead of the id of the previous event.
+    """
+    events, event_id_ = events_from_annotations(raw, event_id=event_id,
+                                                use_rounding=False)
+    durations = raw.annotations.duration
+    durations = np.array(durations * raw.info['sfreq'], int)
```

Comment:
I would use np.around before converting to int to avoid possible loss of precision.

Reply:
This would break backward compatibility. I would put it in a subsequent PR.

Reply:
Hopefully it won't matter since as soon as we release we can eradicate the stim channel synthesis code altogether